refactor/src/Cage.py

```diff
@@ -92,68 +103,98 @@ def update(self, cur_date, step_size, other_farms, reservoir):
+            mortality_rate = sum(pheno_emb) * self.cfg.EMBmort
+            return mortality_rate, pheno_emb, num_susc
+
+        else:
+            return 0, 0, num_susc
+
+    def update_lice_treatment_mortality(self, cur_date):
+        """
+        Calculate the number of lice in each stage killed by treatment.
+        """
+
+        dead_lice_dist = {"L1": 0, "L2": 0, "L3": 0, "L4": 0, "L5m": 0, "L5f": 0}
+
+        mortality_rate, pheno_emb, num_susc = self.get_lice_treatment_mortality_rate(cur_date)
+        susceptible_stages = list(self.lice_population)[2:]
+
+        if mortality_rate > 0:
+            num_dead_lice = np.random.poisson(mortality_rate)
+            num_dead_lice = min(num_dead_lice, num_susc)
+
+            # Now we need to decide how many lice from each stage die,
+            #   the algorithm is to label each louse  1...num_susc
+            #   assign each of these a probability of dying as (phenoEMB)/np.sum(phenoEMB)
+            #   randomly pick lice according to this probability distribution
+            #        now we need to find out which stages these lice are in by calculating the
+            #        cumulative sum of the lice in each stage and finding out how many of our
+            #        dead lice falls into this bin.
+            p = (pheno_emb) / np.sum(pheno_emb)
+            dead_lice = np.random.choice(range(num_susc), num_dead_lice, p=p,
+                                         replace=False).tolist()
+            total_so_far = 0
+            for stage in susceptible_stages:
+                num_dead = len([x for x in dead_lice if total_so_far <= x <
+                                (total_so_far + self.lice_population[stage])])
+                total_so_far += self.lice_population[stage]
+                if num_dead > 0:
+                    self.lice_population[stage] -= num_dead
+                    dead_lice_dist[stage] = num_dead
+
+            self.logger.debug('      distribution of dead lice on farm {}/cage {} = {}'
+                              .format(self.farm_id, self.id, dead_lice_dist))
+
+            assert num_dead_lice == sum(list(dead_lice_dist.values()))
         return dead_lice_dist
 
+    def get_stage_ages_distrib(self, stage: str, size=15):
+        """
+        Create an age distribution (in days) for the sea lice within a lyfecycle stage.
+        In absence of further data or constraints, we simply assume it's a uniform distribution
+        """
+
+        """
+        p = stats.poisson.pmf(range(size), round(stage_age_max_days/2))
+        return p / np.sum(p)
+        """
+        stage_age_max_days = self.cfg.stage_age_evolutions[stage]
+        p = np.ones((size,))
+        p[size - stage_age_max_days:] = 0
+        return p / np.sum(p)
+
     @staticmethod
-    def get_stage_ages(size: int, min: int, mean: int, development_days=25):
+    def get_evolution_ages(size: int, min: int, mean: int, development_days=25):
```

Comment:
This has been there before your changes but I just spotted it - can we change the name of `min` here? Since it's a built-in method name.